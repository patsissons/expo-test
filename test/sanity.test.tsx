import React from 'react';
import { create } from 'react-test-renderer';

describe('react-test-renderer', () => {
  it('can create render output', () => {
    // Arrange
    const component = (<div />);

    // Act
    const result = create(component);

    // Assert
    expect(result).toBeTruthy();
  });

  it('can create a json representation', () => {
    // Arrange
    const component = (<div />);

    // Act
    const result = create(component).toJSON();

    // Assert
    expect(result).toBeTruthy();
    expect(result).toEqual({ type: 'div', props: {}, children: null });
  });

  it('can create a test instance', () => {
    // Arrange
    const component = (<div />);

    // Act
    const result = create(component).root;

    // Assert
    expect(result).toBeTruthy();
    expect(result.type).toBe('div');
    expect(result.props).toEqual({});
    expect(result.children).toEqual([]);
  });

  it('can match a snapshot', () => {
    // Arrange
    const component = (<div />);

    // Act
    const result = create(component).toJSON();

    // Assert
    expect(result).toMatchSnapshot();
  });
});
