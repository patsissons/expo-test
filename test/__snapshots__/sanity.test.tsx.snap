// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-test-renderer can match a snapshot 1`] = `<div />`;
